de.tudarmstadt.ukp.wikipedia.parser/src/main/java/de/tudarmstadt/ukp/wikipedia/parser/html/HtmlWriter.java

```diff
@@ -491,7 +491,7 @@ public static void writeFile(String filename, String encoding, String text) {
         try (Writer destFile = new BufferedWriter(new OutputStreamWriter(new BufferedOutputStream(new FileOutputStream(outFile)), encoding))) {
             destFile.write(text);
         } catch (UnsupportedEncodingException e1) {
-            logger.error("Unsupported encoding exception while opening file " + outFile.getAbsolutePath());
+            logger.error("Unsupported encoding exception while opening file " + outFile.getAbsolutePath(), e1);
```

Comment:
Should be using a placeholder (`{}`) in the log string here.

Reply:
fixed with next commit.

Reply:
Sadly, `org.slf4j.Logger` does not allow us to do both: (a) format the log messages AND (b) chain the catched `Throwable` instance. The only chance to print stack traces is to use


>  /**
     * Log an exception (throwable) at the ERROR level with an
     * accompanying message.
     *
     * @param msg the message accompanying the exception
     * @param t   the exception (throwable) to log
     */
    public void error(String msg, Throwable t);

All other `error(..)` methods do allow parameter replacement, yet NO printing of the stacktrace.

For this reason, we don't have other options than building the String via concatenation.

@reckart If you see another method/way to achieve *both* at once, pls provide your feedback here and I will look for it.

Reply:
If the last parameter in the variadic arguments of an SLF4J logging call is a throwable, then it will be used to log the stack trace. E.g.

```
log.error("Could not read file {}", filename, ioexception);
```

Reply:
Taken from what source / javadoc?

Reply:
https://www.slf4j.org/faq.html#paramException

Reply:
That helps! Right, so I will change it accordingly. Nevertheless, the JavaDoc of these methods does not mention this behaviour... SLF4J guys should improve it to better support other devs.

Reply:
Fixed with next commit.